src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs

```diff
@@ -583,8 +583,10 @@ private static void UpdateExpiration(HttpConnectionContext connection, HttpConte
 
             if (authenticateResultFeature is not null)
             {
-                connection.AuthenticationExpiration =
-                    authenticateResultFeature.AuthenticateResult?.Properties?.ExpiresUtc ?? DateTimeOffset.MaxValue;
+                // get the future Environment.TickCount64 value of when auth expires
+                var expiresTime = (authenticateResultFeature.AuthenticateResult?.Properties?.ExpiresUtc ?? DateTimeOffset.MaxValue);
```

Comment:
If the public API is intended to give control over the `DateTimeOffset` of expiration, we should probably continue repeatedly comparing `DateTimeOffset`s.

`connection.AuthenticationExpiration < utcNow` seems more correct than precomputing the ticks if the expiration was defined as a point of time rather than elapsed time.

Reply:
Updated